core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go

```diff
@@ -459,6 +459,38 @@ func (couchInstance *couchInstance) retrieveApplicationDBNames() ([]string, erro
 	return applicationsDBNames, nil
 }
 
+// retrieveDBNamesMatchedForChannel returns existing db names whose channel segment matches
+// either the original channel name or the truncated channel name.
+// Note that false matching is possible if the channel name exceeds chainNameAllowedLength
+// because another channel may be truncated to the same name. Therefore, this method may
+// may return more than the channel's DB names.
+func (couchInstance *couchInstance) retrieveDBNamesMatchedForChannel(channelName string) (map[string]struct{}, error) {
```

Comment:
Is this function needed?

Instead of allowing the false matching while trying to approximately fetch DBs associated with a channel, can't we simply use `retrieveApplicationDBNames()` at the `initChannelMetadata()` and remove this function. The logic in `initChannelMetadata()` takes care of finding the correct mapping between the namespace and db. This additional complexity does not seem necessary.

Reply:
Since we need to build a map to look up DBNames, this function builds the map with minimum entries to make dbname lookup more efficient. However, because this is one time call for v2.0/2.0 peers only, I agree we don't need the additional complexity.